src/lib.rs

```diff
@@ -630,55 +676,120 @@ impl<'arg, A: Argument + 'arg, I: Iterator<Item = A>> Options<A, I> {
     ///
     /// assert_eq!(args.next(), Some("one"));
     /// assert_eq!(args.next(), Some("two"));
     /// assert_eq!(args.next(), None);
     /// ```
-    pub fn args(&mut self) -> ArgIterator<A, I> {
-        ArgIterator::new(self)
+    pub fn positionals(&mut self) -> Positionals<A, I> {
+        Positionals::new(self)
+    }
+
+    /// Consumes this [`Options`], returning an iterator over the rest
+    /// of the arguments. The returned iterator wraps the one originally
+    /// passed to [`Options::new`].
+    ///
+    /// # Panics
+    ///
+    /// Panics if an option is currently being parsed.
+    pub fn into_positionals(self) -> IntoPositionals<A, I> {
+        match self.state {
+            State::Start { .. } | State::EndOfOption(_) | State::End { .. } => {
+                IntoPositionals::new(None, self.iter)
+            }
+            State::Positional(positional) => IntoPositionals::new(Some(positional), self.iter),
+            _ => {
+                panic!("called Options::into_positionals() while option parsing hasn't finished")
+            }
+        }
+    }
+
+    /// Retrieves the next *argument*. An *argument* is represented by
+    /// [`Arg`] and defined as either an option ([`Opt`]) or a
+    /// positional argument.
+    ///
+    /// This method is implemented entirely using other public methods
+    /// of [`Options`].
+    ///
+    /// # Panics
+    ///
+    /// Panics if [`Options::next_opt`] does.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use getargs::{Arg, Opt, Options};
+    /// #
+    /// let args = ["--option", "VALUE", "arg", "arg2", "-vvv", "--option", "VALUE", "--", "arg3", "arg4", "--not-an-option"];
+    /// let mut opts = Options::new(args.into_iter());
+    ///
+    /// // Regular options and values work...
+    /// assert_eq!(opts.next_arg(), Ok(Some(Arg::Long("option"))));
+    /// assert_eq!(opts.value(), Ok("VALUE"));
+    ///
+    /// // Positional arguments after work...
+    /// assert_eq!(opts.next_arg(), Ok(Some(Arg::Positional("arg"))));
+    /// assert_eq!(opts.next_arg(), Ok(Some(Arg::Positional("arg2"))));
+    ///
+    /// // Options after positional arguments also work
+    /// assert_eq!(opts.next_arg(), Ok(Some(Arg::Short('v'))));
+    /// assert_eq!(opts.next_arg(), Ok(Some(Arg::Short('v'))));
+    /// assert_eq!(opts.next_arg(), Ok(Some(Arg::Short('v'))));
+    /// assert_eq!(opts.next_arg(), Ok(Some(Arg::Long("option"))));
+    /// assert_eq!(opts.value(), Ok("VALUE"));
+    ///
+    /// // `--` is recognized and handled correctly
+    /// assert_eq!(opts.next_arg(), Ok(Some(Arg::Positional("arg3"))));
+    /// assert_eq!(opts.next_arg(), Ok(Some(Arg::Positional("arg4"))));
+    /// assert_eq!(opts.next_arg(), Ok(Some(Arg::Positional("--not-an-option"))));
+    /// assert_eq!(opts.next_arg(), Ok(None));
+    /// #
+    /// # assert_eq!(opts.is_empty(), true);
+    /// ```
+    pub fn next_arg(&'_ mut self) -> Result<A, Option<Arg<A>>> {
```

Comment:
I think these methods should be put in a more sensible order. That way, the most important methods appear at the top in rustdoc. Perhaps:

```
new
next_arg
next_opt
next_positional
value
value_opt
positionals
into_positionals
is_empty
opts_ended
reset_opts_ended
restart
```

Reply:
I actually did reorder them a fair bit but I ordered them in terms of declaration/dependency rather than importance. I'll try importance

Reply:
Here's what I'm going with for now:

1. `next_opt` and `next_arg` in that order, due to `Arg` being a superset of `Opt` - like a natural progression
2. `value` and `value_opt` directly after those, as they are more relevant to `Opt`s and `Arg`s
3. `next_positional`, `positionals`, and `into_positionals`
4. `opts_ended` and `reset_opts_ended`
5. `is_empty` and `restart`